datafusion/common/src/scalar.rs

```diff
@@ -676,13 +696,13 @@ macro_rules! build_values_list {
                     ScalarValue::$SCALAR_TY(None) => {
                         builder.values().append_null();
                     }
-                    _ => panic!("Incompatible ScalarValue for list"),
```

Comment:
I agree this error can be unreachable if invariant is established that the `data_type` and `values` arguments passed to [`new_list`](https://github.com/apache/arrow-datafusion/blob/main/datafusion/common/src/scalar.rs#L1664) align with each other. 
I will change this back to `panic` and thus making `new_list` panic instead of returning `Result`.